File: flutter_app/lib/models/category.dart
class Category {
  final String id;
  final String name;
  final String? description;
  final String? color;
  final String? icon;
  final int sortOrder;
  final bool active;

  Category({
    required this.id,
    required this.name,
    this.description,
    this.color,
    this.icon,
    this.sortOrder = 0,
    this.active = true,
  });

  factory Category.fromJson(Map<String, dynamic> json) => Category(
        id: json['id'],
        name: json['name'],
        description: json['description'],
        color: json['color'],
        icon: json['icon'],
        sortOrder: json['sortOrder'] ?? 0,
        active: json['active'] ?? true,
      );

  Map<String, dynamic> toJson() => {
        'id': id,
        'name': name,
        'description': description,
        'color': color,
        'icon': icon,
        'sortOrder': sortOrder,
        'active': active,
      };
}

File: backend/src/main/java/com/sweetpos/entity/Category.java
package com.sweetpos.entity;

import jakarta.persistence.*;
import lombok.*;

import java.util.ArrayList;
import java.util.List;

@Entity
@Table(name = "categories")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Category extends BaseEntity {

    @Column(nullable = false, length = 100)
    private String name;

    @Column(length = 500)
    private String description;

    @Column(length = 10)
    private String color;

    @Column(length = 10)
    private String icon;

    @Column(nullable = false)
    @Builder.Default
    private int sortOrder = 0;

    @Column(nullable = false)
    @Builder.Default
    private boolean active = true;

    @OneToMany(mappedBy = "category", fetch = FetchType.LAZY)
    @Builder.Default
    private List<Product> products = new ArrayList<>();
}
